core/include/cubos/core/reflection/traits/inherits.hpp

```diff
@@ -0,0 +1,48 @@
+/// @file
+/// @brief Class @ref cubos::core::reflection::InheritsTrait.
+/// @ingroup core-reflection
+
+#include <cubos/core/reflection/type.hpp>
+
+namespace cubos::core::reflection
+{
+    /// @brief Provides inheritance relationship between types.
+    /// @see See @ref examples-core-reflection-traits-inherits for an example of using this trait.
+    /// @ingroup core-reflection
+    class InheritsTrait
+    {
+    public:
+        ~InheritsTrait() = default;
+
+        /// @brief Constructs.
+        InheritsTrait(const Type& type)
+            : mType(&type)
+        {
+        }
+
+        /// @brief Adds an inheritance relationship to the type.
+        /// @tparam T The parent type from which this type inherits.
+        /// @return Trait.
+        template <typename T>
+        static InheritsTrait from()
+        {
+            return InheritsTrait(reflect<T>());
+        }
+
+        /// @brief Checks if the type inherits from the specified parent type.
+        /// @tparam T The parent type to check for inheritance.
+        /// @return Whether type inherits from the specified parent type, false otherwise.
+        template <typename T>
+        bool inherits() const
+        {
+            return mType->is<T>() || (mType->has<InheritsTrait>() && mType->get<InheritsTrait>().inherits<T>());
+        }
+
+        /// @brief Gets the base type of the inherited type.
+        /// @return Type.
+        const Type* base();
```

Comment:
```suggestion
        const Type& base();
```
We usually always use references in the public facing methods as these types may never be null.